rocks-lib/src/lua_package/version.rs

```diff
@@ -1,15 +1,133 @@
+            PackageVersion::Dev { name } => f.write_str(name.as_str()),
+        }
+    }
+}
+
+impl PartialOrd for PackageVersion {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for PackageVersion {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (self, other) {
+            (PackageVersion::SemVer { version: a }, PackageVersion::SemVer { version: b }) => {
+                a.cmp(b)
+            }
+            (PackageVersion::SemVer { .. }, PackageVersion::Dev { .. }) => Ordering::Less,
+            (PackageVersion::Dev { .. }, PackageVersion::SemVer { .. }) => Ordering::Greater,
+            (PackageVersion::Dev { name: a }, PackageVersion::Dev { name: b }) => a.cmp(b),
+        }
+    }
+}
+
+impl FromStr for PackageVersion {
+    type Err = eyre::Error;
+
+    fn from_str(text: &str) -> Result<Self, Self::Err> {
+        let regex: Regex = Regex::new(r"^(dev|scm|git)$").unwrap();
+        if regex.is_match(text) {
+            return Ok(PackageVersion::Dev { name: text.into() });
+        }
+        Ok(PackageVersion::SemVer {
+            version: parse_version(text)?,
+        })
+    }
+}
+
+/// **SemVer version** requirement as defined by <https://semver.org>.
+/// or a **Dev** version requirement, which can be one of "dev", "scm", or "git"
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub enum PackageVersionReq {
+    SemVer { version_req: VersionReq },
+    Dev { name_req: String },
+}
+
+impl PackageVersionReq {
+    pub fn parse(text: &str) -> Result<Self> {
+        PackageVersionReq::from_str(text)
+    }
+    pub fn matches(&self, version: &PackageVersion) -> bool {
+        match (self, version) {
+            (PackageVersionReq::SemVer { version_req }, PackageVersion::SemVer { version }) => {
+                version_req.matches(version)
+            }
+            (PackageVersionReq::SemVer { .. }, PackageVersion::Dev { .. }) => true,
+            (PackageVersionReq::Dev { .. }, PackageVersion::SemVer { .. }) => false,
+            (PackageVersionReq::Dev { name_req }, PackageVersion::Dev { name }) => {
+                name_req.ends_with(name)
+            }
+        }
+    }
+}
+
+impl Display for PackageVersionReq {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PackageVersionReq::SemVer { version_req } => version_req.fmt(f),
+            PackageVersionReq::Dev { name_req } => f.write_str(name_req.as_str()),
+        }
+    }
+}
+
+impl Default for PackageVersionReq {
+    fn default() -> Self {
+        PackageVersionReq::SemVer {
+            version_req: VersionReq::default(),
+        }
+    }
+}
+
+impl FromStr for PackageVersionReq {
+    type Err = eyre::Error;
+
+    fn from_str(text: &str) -> Result<Self, Self::Err> {
+        let regex: Regex = Regex::new(r"^==\s*(dev|git|scm)|\b(dev|git|scm)\b$").unwrap();
+        if regex.is_match(text) {
```

Comment:
suggestion: same here, a collection of `trim_start_matches`, `trim` and `matches!()` could probably supersede invoking an entire regex engine here, unless I'm misinterpreting certain edge cases of this regex.